src/backend/distributed/deparser/deparse_role_stmts.c

```diff
@@ -349,7 +350,20 @@ AppendGrantRoleStmt(StringInfo buf, GrantRoleStmt *stmt)
 {
 	appendStringInfo(buf, "%s ", stmt->is_grant ? "GRANT" : "REVOKE");
 
-#if PG_VERSION_NUM < PG_VERSION_16
+#if PG_VERSION_NUM >= PG_VERSION_16
+	if (!stmt->is_grant)
+	{
+		DefElem *opt = NULL;
+		foreach_ptr(opt, stmt->opt)
+		{
+			if (strcmp(opt->defname, "admin") == 0)
+			{
+				appendStringInfo(buf, "ADMIN OPTION FOR ");
```

Comment:
should we add a simple test for this? To make sure we can properly send the command ?

Reply:
We already have tests for this:
https://github.com/citusdata/citus/blob/main/src/test/regress/sql/create_role_propagation.sql#L69